src/integration-test/java/com/esri/arcgisruntime/toolkit/BookmarksListIntegrationTest.java

```diff
@@ -0,0 +1,320 @@
+        // given a scene view containing a scene with bookmarks
+        SceneView sceneView = new SceneView();
+        Platform.runLater(() -> stackPane.getChildren().add(sceneView));
+
+        ArcGISScene scene = new ArcGISScene(Basemap.createImagery());
+        scene.getBookmarks().add(new Bookmark("Guitar-shaped trees", new Viewpoint(-33.867886, -63.985, 4e4)));
+        scene.getBookmarks().add(new Bookmark("Grand Prismatic Spring", new Viewpoint(44.525049, -110.83819, 6e3)));
+        sceneView.setArcGISScene(scene);
+
+        // when the bookmarks view is added with the scene view
+        BookmarksList bookmarksList = new BookmarksList(sceneView);
+        bookmarksList.setMaxSize(100, 100);
+        StackPane.setAlignment(bookmarksList, Pos.TOP_RIGHT);
+        StackPane.setMargin(bookmarksList, new Insets(10));
+        Platform.runLater(() -> stackPane.getChildren().add(bookmarksList));
+
+        sleep(3000);
+
+        // every bookmark's name will be displayed in the view
+        scene.getBookmarks().forEach(bookmark -> clickOn(bookmark.getName()));
+    }
+
+    /**
+     * Tests that adding and removing bookmarks from the map updates the list.
+     */
+    @Test
+    public void adding_and_removing_bookmarks() {
+        // given a map view containing a map with bookmarks and a bookmarks list
+        MapView mapView = new MapView();
+        Platform.runLater(() -> stackPane.getChildren().add(mapView));
+
+        ArcGISMap map = new ArcGISMap(Basemap.createImagery());
+        Bookmark bookmarkToKeep = new Bookmark("Guitar-shaped trees", new Viewpoint(-33.867886, -63.985, 4e4));
+        Bookmark bookmarkToRemove = new Bookmark("Grand Prismatic Spring", new Viewpoint(44.525049, -110.83819, 6e3));
+        map.getBookmarks().add(bookmarkToKeep);
+        map.getBookmarks().add(bookmarkToRemove);
+        mapView.setMap(map);
+
+        BookmarksList bookmarksList = new BookmarksList(mapView);
+        bookmarksList.setMaxSize(100, 100);
+        StackPane.setAlignment(bookmarksList, Pos.TOP_RIGHT);
+        StackPane.setMargin(bookmarksList, new Insets(10));
+        Platform.runLater(() -> stackPane.getChildren().add(bookmarksList));
+
+        sleep(3000);
+
+        // after adding and removing some bookmarks
+        Bookmark bookmarkToAdd = new Bookmark("Strange Symbol", new Viewpoint(37.401573, -116.867808, 6e3));
+        map.getBookmarks().remove(bookmarkToRemove);
+        map.getBookmarks().add(bookmarkToAdd);
+
+        WaitForAsyncUtils.waitForFxEvents();
+
+        // the bookmark to keep and the added one will be listed and the removed one will not
+        clickOn(bookmarkToKeep.getName());
+        clickOn(bookmarkToAdd.getName());
+        Assertions.assertThrows(FxRobotException.class, () -> clickOn(bookmarkToRemove.getName()));
+    }
+
+    /**
+     * Tests that the list updates when the map is changed on the map view.
+     */
+    @Test
+    public void change_map() {
```

Comment:
This test does not pass at the moment. AFAIK, there is no hook to detect when the map on a map view is changed in the API. We can either release this component without that support, or we can release it whenever we get that capability from the API.